Panic when trying to assign return value of a function returning void

```
package main

func foo()() {
}

func main()() {
	var i i32
	i = foo()
}
```
output:
```
panic: runtime error: index out of range

goroutine 1 [running, locked to thread]:
github.com/skycoin/cx/cxgo/actions.Assignment(0xc42000e1f0, 0x1, 0x1, 0xc4203f5550, 0x1, 0xc42000e208, 0x1, 0x1, 0x1, 0x0, ...)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/assignment.go:170 +0x2166
main.yyParse(0x976a60, 0xc420478a50, 0xc420478a50)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.go:2192 +0x8445
main.main()
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.nn.go:8584 +0x1640
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@
   * #76: Using int literal 0 where 0.0 was needed gave no error.
   * #133: Panic when inserting a newline into a string literal
   * #134: Panic when declaring a variable of an unknown type.
+  * #155: Panic when trying to assign return value of a function returning void.
   * #156: Panic when using a function declared in another package without importing the package.
   * #166: Panic when calling a function from another package where the package name alias a local variable name
   * #271: CX floats cannot handle exponents

--- a/cxgo/actions/assignment.go
+++ b/cxgo/actions/assignment.go
@@ -1,6 +1,8 @@
 package actions
 
 import (
+	"os"
+	
 	. "github.com/skycoin/cx/cx"
 )
 
@@ -103,84 +105,93 @@ func getOutputType(expr *CXExpression) *CXArgument {
 }
 
 // Assignment handles assignment statements with different operators, like =, :=, +=, *=.
-func Assignment(to []*CXExpression, assignOp string, from []*CXExpression) []*CXExpression {
+func Assignment(to []*CXExpression, assignOp string, from []*CXExpression) []*CXExpression {	
 	idx := len(from) - 1
 
-	if pkg, err := PRGRM.GetCurrentPackage(); err == nil {
+	// Checking if we're trying to assign stuff from a function call
+	// And if that function call actually returns something. If not, throw an error.
+	if from[idx].Operator != nil && len(from[idx].Operator.Outputs) == 0 {
+		println(CompilationError(to[0].Outputs[0].FileName, to[0].Outputs[0].FileLine), "trying to use an outputless operator in an assignment")
+		os.Exit(CX_COMPILATION_ERROR)
+	}
 
-		var expr *CXExpression
+	pkg, err := PRGRM.GetCurrentPackage()
+	if err != nil {
+		panic(err)
+	}
 
-		switch assignOp {
-		case ":=":
-			expr = MakeExpression(nil, CurrentFile, LineNo)
-			expr.Package = pkg
+	var expr *CXExpression
 
-			var sym *CXArgument
-			
-			if from[idx].Operator == nil {
-				// then it's a literal
-				sym = MakeArgument(to[0].Outputs[0].Name, CurrentFile, LineNo).AddType(TypeNames[from[idx].Outputs[0].Type])
-			} else {
-				outTypeArg := getOutputType(from[idx])
+	switch assignOp {
+	case ":=":
+		expr = MakeExpression(nil, CurrentFile, LineNo)
+		expr.Package = pkg
 
-				sym = MakeArgument(to[0].Outputs[0].Name, CurrentFile, LineNo).AddType(TypeNames[outTypeArg.Type])
+		var sym *CXArgument
+		
+		if from[idx].Operator == nil {
+			// then it's a literal
+			sym = MakeArgument(to[0].Outputs[0].Name, CurrentFile, LineNo).AddType(TypeNames[from[idx].Outputs[0].Type])
+		} else {
+			outTypeArg := getOutputType(from[idx])
 
-				if from[idx].IsArrayLiteral {
-					sym.Size = from[idx].Inputs[0].Size
-					sym.TotalSize = from[idx].Inputs[0].TotalSize
-					sym.Lengths = from[idx].Inputs[0].Lengths
-				}
-				if outTypeArg.IsSlice {
-					// if from[idx].Operator.Outputs[0].IsSlice {
-					sym.Lengths = append([]int{0}, sym.Lengths...)
-				}
+			sym = MakeArgument(to[0].Outputs[0].Name, CurrentFile, LineNo).AddType(TypeNames[outTypeArg.Type])
 
-				sym.IsSlice = outTypeArg.IsSlice
-				// sym.IsSlice = from[idx].Operator.Outputs[0].IsSlice
+			if from[idx].IsArrayLiteral {
+				sym.Size = from[idx].Inputs[0].Size
+				sym.TotalSize = from[idx].Inputs[0].TotalSize
+				sym.Lengths = from[idx].Inputs[0].Lengths
 			}
-			sym.Package = pkg
-			sym.PreviouslyDeclared = true
-			sym.IsShortDeclaration = true
-
-			expr.AddOutput(sym)
-
-			for _, toExpr := range to {
-				toExpr.Outputs[0].PreviouslyDeclared = true
-				toExpr.Outputs[0].IsShortDeclaration = true
+			if outTypeArg.IsSlice {
+				// if from[idx].Operator.Outputs[0].IsSlice {
+				sym.Lengths = append([]int{0}, sym.Lengths...)
 			}
 
-			to = append([]*CXExpression{expr}, to...)
-		case ">>=":
-			expr = MakeExpression(Natives[OP_UND_BITSHR], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "<<=":
-			expr = MakeExpression(Natives[OP_UND_BITSHL], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "+=":
-			expr = MakeExpression(Natives[OP_UND_ADD], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "-=":
-			expr = MakeExpression(Natives[OP_UND_SUB], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "*=":
-			expr = MakeExpression(Natives[OP_UND_MUL], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "/=":
-			expr = MakeExpression(Natives[OP_UND_DIV], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "%=":
-			expr = MakeExpression(Natives[OP_UND_MOD], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "&=":
-			expr = MakeExpression(Natives[OP_UND_BITAND], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "^=":
-			expr = MakeExpression(Natives[OP_UND_BITXOR], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
-		case "|=":
-			expr = MakeExpression(Natives[OP_UND_BITOR], CurrentFile, LineNo)
-			return ShortAssignment(expr, to, from, pkg, idx)
+			sym.IsSlice = outTypeArg.IsSlice
+			// sym.IsSlice = from[idx].Operator.Outputs[0].IsSlice
 		}
+		sym.Package = pkg
+		sym.PreviouslyDeclared = true
+		sym.IsShortDeclaration = true
+
+		expr.AddOutput(sym)
+
+		for _, toExpr := range to {
+			toExpr.Outputs[0].PreviouslyDeclared = true
+			toExpr.Outputs[0].IsShortDeclaration = true
+		}
+
+		to = append([]*CXExpression{expr}, to...)
+	case ">>=":
+		expr = MakeExpression(Natives[OP_UND_BITSHR], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "<<=":
+		expr = MakeExpression(Natives[OP_UND_BITSHL], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "+=":
+		expr = MakeExpression(Natives[OP_UND_ADD], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "-=":
+		expr = MakeExpression(Natives[OP_UND_SUB], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "*=":
+		expr = MakeExpression(Natives[OP_UND_MUL], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "/=":
+		expr = MakeExpression(Natives[OP_UND_DIV], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "%=":
+		expr = MakeExpression(Natives[OP_UND_MOD], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "&=":
+		expr = MakeExpression(Natives[OP_UND_BITAND], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "^=":
+		expr = MakeExpression(Natives[OP_UND_BITXOR], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
+	case "|=":
+		expr = MakeExpression(Natives[OP_UND_BITOR], CurrentFile, LineNo)
+		return ShortAssignment(expr, to, from, pkg, idx)
 	}
 
 	if from[idx].Operator == nil {

--- a/tests/main.cx
+++ b/tests/main.cx
@@ -401,7 +401,7 @@ func main ()() {
 	runTestEx("issue-141.cx", cx.SUCCESS, "Parser gets confused with `2 -2`", TEST_STABLE, 0)
 	runTest("issue-153.cx", cx.SUCCESS, "Panic in when assigning an empty initializer list to a []i32 variable")
 	runTest("issue-154.cx", cx.SUCCESS, "Cx stack overflow when appending to a slice passed by address")
-	runTestEx("issue-155.cx", cx.COMPILATION_ERROR, "Panic when trying to assign return value of a function returning void", TEST_ISSUE, 0)
+	runTest("issue-155.cx", cx.COMPILATION_ERROR, "Panic when trying to assign return value of a function returning void")
 	runTest("issue-156.cx", cx.COMPILATION_ERROR, "Panic when using a function declared in another package without importing the package")
 	runTest("issue-157.cx", cx.SUCCESS, "Cx memory stomped")
 	runTest("issue-158.cx", cx.SUCCESS, "Invalid offset calculation of non literal strings when appended to a slice")

--- a/tests/testdata/tokens/main.cx.txt
+++ b/tests/testdata/tokens/main.cx.txt
@@ -3124,7 +3124,7 @@ PERIOD
 STRLIT Cx stack overflow when appending to a slice passed by address
 RPAREN 
 SCOLON 
- IDENT runTestEx
+ IDENT runTest
 LPAREN 
 STRLIT issue-155.cx
  COMMA 
@@ -3133,10 +3133,6 @@ PERIOD
  IDENT COMPILATION_ERROR
  COMMA 
 STRLIT Panic when trying to assign return value of a function returning void
- COMMA 
- IDENT TEST_ISSUE
- COMMA 
-INTLIT 0
 RPAREN 
 SCOLON 
  IDENT runTest

